http_api.go

```diff
@@ -0,0 +1,194 @@
+package worker
+
+import (
+	"crypto/subtle"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+// APIHandler handles requests to worker's HTTP API.
+type APIHandler struct {
+	i *CLI
+}
+
+// Setup installs the HTTP routes that will handle requests to the HTTP API.
+func (api *APIHandler) Setup() {
+	api.i.logger.Info("setting up HTTP API")
```

Comment:
This is the only place where `api.i.logger` is accessed, so maybe log this up in `cli.go` when running `Setup`?